### PR TITLE
Add "exp" field to ServerJwt struct

### PR DESCRIPTION
**Pull request by jspam**
Since a jwt update in commit 5032a8303cc263ba7331c5b47807a4da9ff31a15,
this field is required. However, it was only added to the JobJwt struct
in that commit.

Return Result instead of Option for better error messages.

### CODE REVIEW

File: src/bin/sccache-dist/main.rs
Comment (drahnr) [inline src/bin/sccache-dist/main.rs:257]: Shouldn't this be a fixed size type? `u64` or `u128`?

Comment (jspam) [reply src/bin/sccache-dist/main.rs]: Right, changed to `u64` as in `JobJwt`.

**Review by drahnr — APPROVED** [review]
LGTM

**Comment by jspam** [comment]
Could a maintainer please approve the CI run in Github? Thanks.

**Comment by drahnr** [comment]
Done, sorry I forgot that.

**Comment by jspam** [comment]
Apologies if I sound naggy, but is there anything left I need to do? Or just relax and wait for the PR to be merged? :-) (I do not myself have the rights to merge it.)

**Comment by drahnr** [comment]
Lean back and relax :)

**Pull request merged**
